backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs

```diff
@@ -144,6 +159,18 @@ public async Task<AvailabilitySlotDto> UpdateAvailabilitySlot(long id, SaveAvail
                 _context.Remove(availabilitySlot.AdvancedSlotSettings);
             }
 
+            if (updateAvailabilityDto.EmailSettings is not null)
+            {
+                var newTemplate = _mapper.Map<EmailTemplate>(updateAvailabilityDto.EmailSettings);
+                var oldTemplate = await _context.EmailTemplates.Where(t => t.TemplateType == newTemplate.TemplateType && t.AvailabilitySlotId == availabilitySlot.Id).FirstOrDefaultAsync();
+                if (oldTemplate != null)
+                {
+                    _context.EmailTemplates.Remove(oldTemplate);
+                }
+
+                availabilitySlot.EmailTemplates.Add(newTemplate);
```

Comment:
You need to update template, not delete and add

Reply:
We need to have only 1 template with type(Confirmation, Cancellation, etc.) for settings. For this reason previous email template is removing

Reply:
I rewrite code for update previous template